File: DSA/Functions/BinaryToDecimal.java
package Functions;

public class BinaryToDecimal {
    public static void B2D(int n){
        int decimal=0;
        int lastDigit=0;
        int pow=0;
        while(n>0){
            lastDigit=n%10;
            decimal=decimal +  (lastDigit* (int)Math.pow(2,pow));
            pow++;
            n=n/10;
        }
        System.out.println("Decimal of binary number: "+ decimal);
    }
    public static void main(String args[]){
        B2D(101);
    }
}

File: Flow Control, Functions and Loop/BinaryToDecimal.cpp
#include <iostream>

using namespace std;
int binaryToDecimal(int n)
{
    int val=0;
    int lastDigit=0;
    int base=1;
    while(n>0)
    {
        lastDigit=n%10;
        val= val + (lastDigit*base);
        n= n/10;
        base= base*2;
    }
    return val;
}


int main()
{
    int n;
    cout<<"Enter a  number ";
    cin>>n;
    cout<<binaryToDecimal(n);
    return 0;
}
